File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.cpp
// m3hu1
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> stOps;
         for (int i = 0; i < tokens.size(); i++) {
             if (tokens[i] == "+") {
                 int a = stOps.top();
                 stOps.pop();
                 int b = stOps.top();
                 stOps.pop();
                 stOps.push(a+b);
             }
             else if (tokens[i] == "-") {
                 int a = stOps.top();
                 stOps.pop();
                 int b = stOps.top();
                 stOps.pop();
                 stOps.push(b- a);
             }
             else if (tokens[i] == "*") {
                 int a = stOps.top();
                 stOps.pop();
                 int b = stOps.top();
                 stOps.pop();
                 stOps.push(a * b);
             }
             else if(tokens[i] == "/"){
                 int a = stOps.top();
                 stOps.pop();
                 int b = stOps.top();
                 stOps.pop();
                 stOps.push(b / a);
             }
             else {
                 stOps.push(stoi(tokens[i]));
             }

         }

         return stOps.top();
    }
};

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        def sp():
            return stack.pop()
        # just postfix
        stack = []

        for s in tokens:
            if s == "+":
                stack.append(sp() + sp())
            elif s == "-":
                a = sp()
                b = sp()
                stack.append(b - a)
            elif s == "*":
                stack.append(sp() * sp())
            elif s == "/":
                a = sp()
                b = sp()
                stack.append(int(b / a))
            else:
                stack.append(int(s))
            
        return stack[-1]